TUTORIAL.md

```diff
@@ -0,0 +1,587 @@
+  ))}
+</ul>
+```
+
+Magic! Let’s do one more thing–let people vote on these questions!
+
+Open `app/questions/pages/questions/[id].tsx` in your editor. First, we’re going to improve this page somewhat. 
+
+1. Replace `<h1>Question {question.id}</h1>` with `<h1>{question.text}</h1>`.
+
+2. Delete the `pre` element, and copy in our choices list which we wrote before:
+
+
+```jsx
+<ul>
+  {question.choices.map((choice) => (
+    <li key={choice.id}>
+      {choice.text} - {choice.votes} votes
+    </li>
+  ))}
+</ul>
+```
+
+If you go back to your browser, your page should now look something like this!
+
+
+<img width="567" alt="Screenshot 2020-04-27 at 16 06 55" src="https://user-images.githubusercontent.com/24858006/80387990-3c3d8b80-88a1-11ea-956a-5be85f1e8f12.png">
+
+Now that you’ve improved the question page, it’s time for the vote button.
+
+First, we need a new mutation. Create a file at `app/questions/mutations/updateChoice.ts`, and paste in the following code:
+
+```js
+import db, { ChoiceUpdateArgs } from "db"
+
+export default async function updateChoice(args: ChoiceUpdateArgs) {
+  // Don't allow updating ID
+  delete args.data.id
+
+  const choice = await db.choice.update(args)
+
+  return choice
+}
+```
+
+Back in `app/questions/pages/questions/[id].tsx`, we can now add a vote button.
+
+In our `li`, add a button like so:
+
+```jsx
+<li key={choice.id}>
+  {choice.text} - {choice.votes} votes
+  <button>Vote</button>
+</li>
+```
+
+Then, import our `updateChoice` mutation, and create a `handleVote` function in our page:
+
+```jsx
+import updateChoice from "app/questions/mutations/updateChoice"
+
+...
+
+const handleVote = async (id, votes) => {
+  try {
+    const updated = await updateChoice({
+      where: { id },
+      data: { votes: votes + 1 },
+    })
+    alert("Success!" + JSON.stringify(updated))
+  } catch (error) {
+    alert("Error creating question " + JSON.stringify(error, null, 2))
+  }
+}
+
+return (
+
+...
+```
+
+Finally, we’ll just tell our `button` to call that function!
+
+```jsx
+<button onClick={() => handleVote(choice.id, choice.votes)}>Vote</button>
+```
+
+Just to be sure, this is what all that should look like:
+
+```jsx
+import { Suspense } from "react"
+import { Head, Link, useRouter, useQuery } from "blitz"
+import getQuestion from "app/questions/queries/getQuestion"
+import deleteQuestion from "app/questions/mutations/deleteQuestion"
+import updateChoice from "app/questions/mutations/updateChoice"
+
+export const Question = () => {
+  const router = useRouter()
+  const id = parseInt(router?.query.id as string)
+  const [question] = useQuery(getQuestion, { where: { id } })
+
+  const handleVote = async (id, votes) => {
+    try {
+      const updated = await updateChoice({
+        where: { id },
+        data: { votes: votes + 1 },
+      })
+      alert("Success!" + JSON.stringify(updated))
+    } catch (error) {
+      alert("Error creating question " + JSON.stringify(error, null, 2))
+    }
+  }
```

Comment:
Should this be moved out and above the Question function?

Reply:
I usually colocate handler functions within components but more than happy to change this if it’s just a question of code style(?)

Reply:
Ok this is fine